lib/circuit.js

```diff
@@ -222,6 +232,17 @@ class CircuitBreaker extends EventEmitter {
         handleError(error, this, timeout, args, resolve, reject);
       }
     });
+    if (this.options.cache) {
+      CACHE.set(this, value);
```

Comment:
Hmm. This would mean that we are caching failures as well, I suppose. I'm not sure I like that.